src/qililab/instruments/qblox/qblox_module.py

```diff
@@ -551,6 +551,20 @@ def reset(self):
         self.clear_cache()
         self.device.reset()
 
+    def upload_qpysequence(self, qpysequence: QpySequence, port: str):
+        """Upload the qpysequence to its corresponding sequencer.
+
+        Args:
+            qpysequence (QpySequence): The qpysequence to upload.
+            port (str): The port of the sequencer to upload to.
+        """
+        sequencers = self.get_sequencers_from_chip_port_id(chip_port_id=port)
+        for sequencer in sequencers:
+            logger.info("Sequence program: \n %s", repr(qpysequence._program))  # pylint: disable=protected-access
+            self.device.sequencers[sequencer.identifier].sequence(qpysequence.todict())
+            self.device.sequencers[sequencer.identifier].sync_en(True)
+            self.sequences[sequencer.identifier] = (qpysequence, True)
```

Comment:
Do we really need this line?

Reply:
Ah I see you use it for the acquisitions method

Reply:
Shouldn't we then clean this variable after getting the acquisitions? If you upload two QPrograms (in a row) with different sequencers, in the second execution you would get the results of all sequencers!

Reply:
Yes, you are right! I will do that.